ote_sdk/ote_sdk/tests/entities/test_resultset.py

```diff
@@ -0,0 +1,125 @@
+# Copyright (C) 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+from datetime import datetime
+
+import pytest
+
+from ote_sdk.entities.id import ID
+from ote_sdk.entities.metrics import NullPerformance
+from ote_sdk.entities.resultset import ResultSetEntity, ResultsetPurpose
+from ote_sdk.tests.constants.ote_sdk_components import OteSdkComponent
+from ote_sdk.tests.constants.requirements import Requirements
+
+
+@pytest.mark.components(OteSdkComponent.OTE_SDK)
+class TestResultset:
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_resultset_purpose(self):
+        """
+        <b>Description:</b>
+        Check the ResultsetPurpose can correctly return the value
+
+        <b>Input data:</b>
+        Denotes, stages
+
+        <b>Expected results:</b>
+        Test passes if the results match
+        """
+
+        denotes = ["EVALUATION", "TEST", "PREEVALUATION"]
+        stages = ["Validation", "Test", "Pre-validation"]
+
+        resultset_purpose = ResultsetPurpose
+        assert len(resultset_purpose) == 3
+
+        for i in ResultsetPurpose:
+            resultset_purpose = ResultsetPurpose(i)
+            assert repr(resultset_purpose) == denotes[i.value]
+            assert str(resultset_purpose) == stages[i.value]
+
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_resultset_entity(self):
+        """
+        <b>Description:</b>
+        Check the ResultSetEntity can correctly return the value
+
+        <b>Input data:</b>
+        Mock data
+
+        <b>Expected results:</b>
+        Test passes if incoming data is processed correctly
+
+        <b>Steps</b>
+        1. Create dummy data
+        2. Check the processing of default values
+        3. Check the processing of changed values
+        """
+
+        test_data = {
+            "model": None,
+            "ground_truth_dataset": None,
+            "prediction_dataset": None,
+            "purpose": None,
+            "performance": None,
+            "creation_date": None,
+            "id": None,
+        }
+
+        result_set = ResultSetEntity(**test_data)
+
+        for name, value in test_data.items():
```

Comment:
You compare attributes of result_set with `None` always. Could you make "model" to be instance of ModelEntity, "ground_truth_dataset" to be an instance of DatasetEntity and so on?

Reply:
Here, the possibility of assigning an attribute is simply checked. Is it worth complicating the test so much?